core/src/test/java/org/apache/druid/java/util/http/client/pool/ResourcePoolTest.java

```diff
@@ -44,7 +44,8 @@ public void setUp()
     EasyMock.replay(resourceFactory);
```

Comment:
not comment on this line but we should also add tests when lazy connection is set to true and verify that pool size increases when more connections are requested.

Reply:
done by adding relevant tests which attempt to get resources from pools without any existing resources